core/src/main/scala/org/apache/spark/util/collection/ChainedBuffer.scala

```diff
@@ -32,7 +32,7 @@ private[spark] class ChainedBuffer(chunkSize: Int) {
   assert(math.pow(2, chunkSizeLog2).toInt == chunkSize,
```

Comment:
Could you use `java.lang.Long.numberOfTrailingZeros(java.lang.Long.highestOneBit(chunkSize)).toLong` to get `chunkSizeLog2`? I'm not sure if `math.log` is accurate since it's double. And please also update the assertion to `require((1 << chunkSizeLog2) == chunkSize, ...)`.